fix: fix priceInfo 'nil' error on discovery

rickstaa opened the pull request:
@eliteprox this pull request fixes a problem that was still in https://github.com/livepeer/go-livepeer/pull/3059 where when a Gateway send a discovery request a `nil` error was thrown.

#### How to reproduce

I think the error is caused by 

1. Checkout https://github.com/livepeer/go-livepeer/pull/3059.
2. Startup the AI orchestrator without the `pricePerUnit` flag.
3. Startup an AI Gateway.
4. Wait till the Gateway sends a request to the `Get

rickstaa committed:
fix: fix priceInfo 'nil' error on discovery

This commit ensures that when the `transcodePrice` is not set by the AI
orchestrator no `nil` error is thrown when a Gateway requests the
orchestrators OrchInfo.

eliteprox commented:
This looks good. It makes sense for `basePrice` to start with a value than null. Transcode price will be used when `pricePerUnit` is set.

rickstaa committed:
fix(ai): fix incorrect transcodePrice condition

This commit fixes the check that is performed to check if transcodePrice
is set.
